Improve Google Pay troubleshooting with detailed user facing logs

cttsai-stripe opened the pull request:
# Summary
<!-- Simple summary of what was changed. -->
1. Log "Google Pay is not enabled for this session." fo element sessions.
2. Log "GooglePayConfiguration is not set." for `GooglePayConfiguration`
3. Log "Google Pay api check failed. " for Google Pay API

# Motivation
<!-- Why are you making this change? If it's for fixing a bug, if possible, please include a code snippet or example project that demonstrates the issue. -->
https://jira.corp.stripe.com/browse/MOBILESDK-2782

# Testing
<!-- How was the code tested? Be as specific as possible. -->
- [ ] Added tests
- [x] Modified tests
- [ ] Manually verified

# Screenshots
| Before  | After |
| ------------- | ------------- |
| *before screenshot*  | *after screenshot* |

# Changelog
<!-- Is this a notable change that affects users? If so, add a line to `CHANGELOG.md` and prefix the line with one of the following:
    - [Added] for new features.
    - [Changed] for changes in existing functionality.
    - [Deprecated] for soon-to-be removed features.
    - [Removed] for now removed features.
    - [Fixed] for any bug fixes.
    - [Security] in case of vulnerabilities.
-->
- [Added] #10496 Improve Google Pay troubleshooting with detailed user facing logs

cttsai-stripe reviewed (commented):
> - The device has google play services
> - The device has a signed in google account

These are verified in api check

> - The payment_method_types includes card

I am not sure how to test this.

> - The `PaymentSheet.Configuration` has `GooglePayConfiguration`

Logged in `PaymentElementLoader` and tested in `DefaultPaymentElementLoaderTest`

> - The `com.google.android.gms.wallet.api.enabled` metadata is added to manifest

I don't think we can check that in code. This should be pointed out in the doc.

samer-stripe commented on payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayRepository.kt:
```diff
@@ -119,9 +125,25 @@ internal class DefaultGooglePayRepository(
+                    "PaymentsClient#public-taskboolean-isreadytopay-isreadytopayrequest-request) " +
+                    "for more details."
+            )
+        }
```
This will get logged twice if the request itself fails

samer-stripe commented on payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayRepository.kt:
```diff
@@ -106,6 +109,9 @@ internal class DefaultGooglePayRepository(
             logger.error("Google Pay json parsing failed.", it)
+            userFacingLogger.logWarningWithoutPii(
+                "Google Pay json parsing failed."
+            )
```
I think we should avoid logging if json parsing fails to merchants. If something goes wrong, it's likely something wrong with what we are doing. We already have an error event informing us if something goes wrong with JSON parsing.

cttsai-stripe replied:
Deleted

cttsai-stripe commented on paymentsheet/src/test/java/com/stripe/android/paymentsheet/state/DefaultPaymentElementLoaderTest.kt:
```diff
@@ -2517,6 +2522,7 @@ internal class DefaultPaymentElementLoaderTest {
                 clientSecret = PaymentSheetFixtures.PAYMENT_INTENT_CLIENT_SECRET.value,
             ),
             paymentSheetConfiguration = PaymentSheet.Configuration.Builder(merchantDisplayName = "Example, Inc.")
+                .googlePay(ConfigFixtures.GOOGLE_PAY)
```
If missing this line, the logger will contain "google pay not configured" error message, which is not what we want to see/test in this case.

samer-stripe commented:
- Can we consolidate everything into a single message rather than multiple logs?
- Looking at `ElementsSession` instance, you should be able to check if `google_pay` is in the `payment_method_types` on the intent.
- We should also add a message indicating what other steps to take if everything else appears to be available such as check if the device is logged into a Google account and has a payment method attached to the wallet.

cttsai-stripe commented:
> * Can we consolidate everything into a single message rather than multiple logs?

Updated
> * Looking at `ElementsSession` instance, you should be able to check if `google_pay` is in the `payment_method_types` on the intent.

Do we want to check `card` or `google_pay`? if the latter, does `elementsSession.isGooglePayEnabled` already do the work?
> * We should also add a message indicating what other steps to take if everything else appears to be available such as check if the device is logged into a Google account and has a payment method attached to the wallet.

Updated.
